discretize/CurvilinearMesh.py

```diff
@@ -25,7 +27,62 @@ def normalize3D(x):
     return x/np.kron(np.ones((1, 3)), utils.mkvc(length3D(x), 2))
 
 
-class CurvilinearMesh(BaseRectangularMesh, DiffOperators, InnerProducts, CurviView):
+class Array(properties.Array):
```

Comment:
This is the implementation of an array that doesn't care about shape (I allow shape to be `'*'` which could be a 1D, 2D or 3D array). See discussion on aranzgeo/properties#192

Reply:
Union property could work well - see my comment on aranzgeo/properties#192